README.md

```diff
@@ -137,3 +137,60 @@ public class Creditcard
   public string ShowFirstAndLastThreeAndCustomMaskInTheMiddle { get; set; }
 }
 ```
+
+#### Masking a string property with Regular Expressions
+
+Apply the `LogReplaced` attribute on a string, in which you want to apply a RegEx replacement during Logging.
+
+This is applicable in scenarios which a string contains both Sensitive and Non-Sensitive information. An example of this could be a string such as "__Sensitive|NonSensitive__". Then you can apply the attibute like the following snippet:
+```csharp
+[LogReplaced(@"([a-zA-Z0-9]+)\|([a-zA-Z0-9]+)", "***|$2")]
+public property Information { get; set; }
+
+// Will log: "***|NonSensitive"
+``` 
+
+`LogReplaced` attribute is available with two constructors:
+
+```csharp
+LogReplaced(string pattern, string replacement)
+
+LogReplaced(string pattern, string replacement, RegexOptions regexOptions)
+```
+
+ - **Pattern:** The pattern that should be applied on value..
+ - **Replacement:** The string that will be applied by RegEx. 
+ - **RegexOptions:** The [RegexOptions](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regexoptions?view=netcore-3.1) that will be applied. Defaults to __RegexOptions.None__
+
+ **Examples**
+
+```csharp
+public class CreditCard
```

Comment:
We could probably keep this down to only one or two examples, since they're quite similar?